packages/backend/src/clients/Slack/SlackClient.ts

```diff
@@ -580,4 +667,191 @@ export class SlackClient {
+                    },
+                    logLevel: LogLevel.INFO,
+                    port: this.lightdashConfig.slack.port,
+                    socketMode: this.lightdashConfig.slack.socketMode,
+                    appToken: this.lightdashConfig.slack.appToken,
+                });
+                this.addEventListeners(app, unfurlService, aiAgentService);
+            } else {
+                const slackReceiver = new ExpressReceiver({
+                    ...this.getSlackOptions(),
+                    installationStore: {
+                        storeInstallation: (i) =>
+                            this.slackAuthenticationModel.createInstallation(i),
+                        fetchInstallation: (i) =>
+                            this.slackAuthenticationModel.getInstallation(i),
+                        deleteInstallation: (i) =>
+                            this.slackAuthenticationModel.deleteInstallation(i),
+                    },
+                    logLevel: LogLevel.INFO,
+                    app: expressApp,
+                });
+                const app = new App({
+                    ...this.getSlackOptions(),
+
+                    receiver: slackReceiver,
+                });
+                this.addEventListeners(app, unfurlService, aiAgentService);
+            }
+        } catch (e: unknown) {
+            Logger.error(`Unable to start Slack app ${e}`);
+        }
+    }
+
+    protected addEventListeners(
+        app: App,
+        unfurlService: UnfurlService,
+        // TODO: aiAgentService is optional, but we need to handle it in the service
+        aiAgentService: AnyType,
+    ) {
+        app.event('link_shared', (m) => this.unfurlSlackUrls(m, unfurlService));
+    }
+
+    private async sendUnfurl(
+        event: AnyType,
+        originalUrl: string,
+        unfurl: Unfurl,
+        client: AnyType,
+    ) {
+        const unfurlBlocks = getUnfurlBlocks(originalUrl, unfurl);
+        await client.chat
+            .unfurl({
+                ts: event.message_ts,
+                channel: event.channel,
+                unfurls: unfurlBlocks,
+            })
+            .catch((e: unknown) => {
+                this.analytics.track({
+                    event: 'share_slack.unfurl_error',
+                    userId: event.user,
+                    properties: {
+                        error: `${getErrorMessage(e)}`,
+                    },
+                });
+                Logger.error(
+                    `Unable to unfurl on slack ${JSON.stringify(
+                        unfurlBlocks,
+                    )}: ${JSON.stringify(e)}`,
+                );
+            });
+    }
+
+    // TODO: this is an antipattern, we should not have UnfurlService as a parameter here
+    // in fact, we should not have unfurlSlackUrls here at all, it should be moved to the service
+    async unfurlSlackUrls(message: AnyType, unfurlService: UnfurlService) {
```

Comment:
would be nice to have `message` as `SlackEventMiddlewareArgs<'link_shared'> &
            AllMiddlewareArgs<StringIndexed>` but I understand that `teamId` below would have to be handled when not defined so that could be an issue to get this out 🤔

Reply:
Yeah, I agree. The function was moved as it was, I haven't updated that part